drivers/sx127x/sx127x_internal.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2016 Unwired Devices <info@unwds.com>
+ *               2017 Inria Chile
+ *               2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx127x
+ * @{
+ * @file
+ * @brief       implementation of internal functions for sx127x
+ *
+ * @author      Eugene P. <ep@unwds.com>
+ * @author      José Ignacio Alamos <jose.alamos@inria.cl>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "irq.h"
+
+#include "sx127x.h"
+#include "sx127x_registers.h"
+#include "sx127x_internal.h"
+#include "sx127x_params.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+bool sx127x_test(sx127x_t *dev)
+{
+    /* Read version number and compare with sx127x assigned revision */
+    uint8_t version = sx127x_reg_read(dev, SX127X_REG_VERSION);
+
+    if (version == VERSION_SX1272) {
+        DEBUG("SX1272 transceiver type detected.\n");
+        dev->driver_type = SX1272_TYPE;
+    }
+    else if (version == VERSION_SX1276) {
+        DEBUG("SX1276 transceiver type detected.\n");
+        dev->driver_type = SX1276_TYPE;
+    }
+    else {
+        DEBUG("[Error] sx127x test failed, invalid version number: %d\n",
+                  version);
+        return false;
+    }
+
+    return true;
+}
+
+uint32_t sx127x_random(sx127x_t *dev)
+{
+    uint8_t i;
+    uint32_t rnd = 0;
+
+    sx127x_set_modem(dev, SX127X_MODEM_LORA); /* Set LoRa modem ON */
+
+    /* Disable LoRa modem interrupts */
+    sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGSMASK, SX127X_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                     SX127X_RF_LORA_IRQFLAGS_RXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                     SX127X_RF_LORA_IRQFLAGS_VALIDHEADER |
+                     SX127X_RF_LORA_IRQFLAGS_TXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_CADDONE |
+                     SX127X_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                     SX127X_RF_LORA_IRQFLAGS_CADDETECTED);
+
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_STANDBY);
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, SX127X_RNG_REG_MODEM_CONFIG1);
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, SX127X_RNG_REG_MODEM_CONFIG2);
+
```

Comment:
SX127X_REG_LR_MODEMCONFIG2 otherwise you rewrite the same register.
Watch out : SX127X_RNG_REG_MODEM_CONFIG (1&2) are different for SX1272 & SX1276